### DISCUSSION TITLE
How to use Ruby 3.4 `GC.config(full_mark: false)`?

### DISCUSSION THREAD

**Discussion by PikachuEXE** [puma/puma#3580] [General]
Issue on Ruby tracker https://bugs.ruby-lang.org/issues/20443
Doc https://docs.ruby-lang.org/en/master/GC.html
Article `Next Generation Out of Band Garbage Collection` https://railsatscale.com/2024-10-23-next-generation-oob-gc/

Asking in advance

**Comment by dentarg** [comment]
The `out_of_band` hook: https://www.rubydoc.info/gems/puma/Puma%2FDSL:out_of_band

https://github.com/puma/puma/blob/4d7f4e77bde20df51bfad4d8a1f290273c6b1fe7/lib/puma/dsl.rb#L928-L940

(Referenced both by the [Ruby ticket](https://bugs.ruby-lang.org/issues/20443) and the [Rails issue](https://github.com/rails/rails/issues/50449))

**Reply by PikachuEXE** [reply]
Thanks. Did see puma in the ticket but `out_of_band` is way below so I don't see it (and I forgot `out_of_band` is a thing in puma

**Reply by PikachuEXE** [reply]
Used in `puma.rb`:
```ruby
# Available since 3.4
oobgc_available = GC.respond_to?(:config)
if oobgc_available
  on_worker_boot do
    GC.config(rgengc_allow_full_mark: false)
  end
  out_of_band do
    if GC.latest_gc_info(:need_major_by)
      GC.start
    end
  end
end
```

https://github.com/PikaSer-Cosmos/likecoin-chain-tx-indexer-pika/commit/26b795ab98f2178a550ae749025ec8d51051c7f0
